src/core/libraries/usbd/usbd.cpp

```diff
@@ -44,7 +44,16 @@ void PS4_SYSV_ABI sceUsbdExit() {
 s64 PS4_SYSV_ABI sceUsbdGetDeviceList(SceUsbdDevice*** list) {
     LOG_DEBUG(Lib_Usbd, "called");
 
-    return libusb_to_orbis_error(libusb_get_device_list(g_libusb_context, list));
+    ssize_t count = libusb_get_device_list(g_libusb_context, list);
+    if (count < 0)
+        return libusb_to_orbis_error((int)count);
+
+    if (count > 2)
+        LOG_WARNING(Lib_Usbd, "Too Many USB Devices Connected");
+    libusb_unref_device((*list)[2]);
+    count = 0;
```

Comment:
```suggestion
    if (count > 2) {
        LOG_WARNING(Lib_Usbd, "Too many USB devices connected to list: {}", count);
        for (auto i = 0; i < count; i++) {
            libusb_unref_device((*list)[i]);
        }
        count = 2;
    }
```

Seems to me that if we're just unreferencing devices past the end, we should a) dereference all of them, not just the one at index 2, and b) set the count to 2 to let the game see the remaining devices we did not unreference.

Reply:
So after several testing, i had to return -1 as an error so it will not have any memleaks, i spend last night 40 minutes on main menu no issues, but passing count as 1-2 creates a leak, i tried to do freedevice too and it didnt pass booting into main menu, stucked there. I can do more testing later on this week i need  to dump some games. feel free to propuse any other change i can test it and see how it behaves.

Reply:
with the last division i can confirm no memory leaks, and also is set to 2 instead of -1